File: comics.go
package comicshelf

import (
	"context"
	"time"
)

type Comic struct {
	Id           int    `json:"id"`
	Title        string `json:"title"`
	Urls         []Url  `json:"urls"`
	Thumbnail    string `json:"thumbnail"`
	Format       string `json:"format"`
	IssuerNumber int    `json:"issuer_number"`
	OnSaleDate   string `json:"on_sale_date"`
	Attribution  string `json:"attribution"`
	SeriesId     int    `json:"series_id"`
}

type ComicService interface {
	GetWeeklyComics(ctx context.Context, t time.Time) (Page[Comic], error)
	GetComic(ctx context.Context, id int) (Comic, error)
}

File: src/comics/comics.ts
export class Page<T> {
  constructor(
    readonly limit: number,
    readonly total: number,
    readonly count: number,
    readonly offset: number,
    readonly results: T[],
  ) {}
}

export class Url {
  constructor(
    readonly type: string,
    readonly url: string,
  ) {}
}

export class Comic {
  constructor(
    readonly id: number,
    readonly title: string,
    readonly urls: Url[],
    readonly thumbnail: string,
    readonly format: string,
    readonly issuer_number: number,
    readonly on_sale_date: string,
    readonly attribution: string,
    readonly attribution_link: string,
    readonly series_id: string,
  ) {}
}
